src/FakeItEasy/Configuration/BuildableCallRule.cs

```diff
@@ -160,6 +160,24 @@ public virtual void ApplyWherePredicate(Func<IFakeObjectCall, bool> predicate, A
 
         public abstract void UsePredicateToValidateArguments(Func<ArgumentCollection, bool> predicate);
 
+        /// <summary>
+        /// Clones the part of the rule that describes which call is being configured.
+        /// </summary>
+        /// <returns>The cloned rule.</returns>
+        public BuildableCallRule CloneCallSpecification()
+        {
+            var clone = this.CloneCallSpecificationCore();
+            clone.wherePredicates.AddRange(this.wherePredicates);
+            return clone;
+        }
+
+        /// <summary>
+        /// When overridden in a derived class, returns a new instance of the same type and copies the call
+        /// specification members for that type.
+        /// </summary>
+        /// <returns>The cloned rule.</returns>
+        protected abstract BuildableCallRule CloneCallSpecificationCore();

```

Comment:
Yeah, I guess CRTP is what I had in mind. Let's leave it as-is. It's internal, so it's only contributors which have to follow that specific doc, and I'm sure we can manage, given our current tempo of creating new types that inherit from `BuildableCallRule`.